tests/unit/h/services/jwt_test.py

```diff
@@ -28,56 +44,285 @@ def test_decode_token(self, service, jwt, PyJWKClient):
+            service.decode_oidc_idtoken(
                 sentinel.token, sentinel.key_set_url, sentinel.algorithms
             )
 
-    def test_decode_token_with_invalid_jwt_header(self, service, jwt):
+    def test_decode_oidc_idtoken_with_invalid_jwt_header(self, service, jwt):
         jwt.get_unverified_header.side_effect = InvalidTokenError()
 
-        with pytest.raises(TokenValidationError, match="Invalid JWT"):
-            service.decode_token(
+        with pytest.raises(JWTDecodeError, match="Invalid JWT"):
+            service.decode_oidc_idtoken(
                 sentinel.token, sentinel.key_set_url, sentinel.algorithms
             )
 
-    def test_decode_token_with_invalid_jwt(self, service, jwt):
+    def test_decode_oidc_idtoken_with_invalid_jwt(self, service, jwt):
         jwt.decode.side_effect = [{"aud": "AUD", "iss": "ISS"}, InvalidTokenError()]
 
-        with pytest.raises(TokenValidationError, match="Invalid JWT"):
-            service.decode_token(
+        with pytest.raises(JWTDecodeError, match="Invalid JWT"):
+            service.decode_oidc_idtoken(
                 sentinel.token, sentinel.key_set_url, sentinel.algorightms
             )
 
-    @pytest.fixture(autouse=True)
-    def jwt(self, patch):
-        return patch("h.services.jwt.jwt")
+    @dataclass
+    class JWTPayload:
+        foo: str
 
-    @pytest.fixture(autouse=True)
-    def PyJWKClient(self, patch):
-        return patch("h.services.jwt.PyJWKClient")
+    def test_encode_decode_symmetric(self, service, payload):
+        token = service.encode_symmetric(
+            payload,
+            expires_in=timedelta(hours=1),
+            issuer="test_issuer",
+            audience="test_audience",
+        )
+        decoded_payload = service.decode_symmetric(
+            token,
+            audience="test_audience",
+            payload_class=TestJWTService.JWTPayload,
+        )
+
+        assert decoded_payload == payload
+
+    def test_encode_symmetric_required_kwargs(self, service, payload):
+        # The expires_in, issuer and audience arguments are required:
+        # encode_symmetric() doesn't support encoding JWTs without these
+        # security and debugging features, and decode_symmetric() wouldn't
+        # decode one.
+        with pytest.raises(
+            TypeError,
+            match=r"^JWTService\.encode_symmetric\(\) missing 3 required keyword-only arguments: 'expires_in', 'issuer', and 'audience'$",
+        ):
+            service.encode_symmetric(payload)
+
+    def test_encode_symmetric_keyword_only_args(self, service, payload):
+        # To prevent any confusion or mixups expires_in, issuer and
+        # audience are keyword-only arguments (payload is the only positional
+        # argument).
+        with pytest.raises(
+            TypeError,
+            match=r"^JWTService\.encode_symmetric\(\) takes 2 positional arguments but 5 were given$",
```

Comment:
Usually I would consider it redundant to check things in tests that are enforced via language syntax or types in the interface. For types it depends on how sure you can be that calls will be type-checked. I suppose you could argue for an exception here on the basis this is security sensitive.